custom/icds_reports/reports/awc_daily_status.py

```diff
@@ -138,16 +138,17 @@ def get_awc_daily_status_data_chart(domain, config, loc_level, show_test=False):
         data['open_in_day'][date_in_miliseconds]['y'] += in_day
         data['launched'][date_in_miliseconds]['y'] += valid
 
-    top_locations = sorted(
-        [
-            dict(
-                loc_name=key,
-                percent=value['in_day'] * 100 / (value['all'] or 1)
-            ) for key, value in six.iteritems(best_worst)
-        ],
-        key=lambda x: x['percent'],
-        reverse=True
-    )
+    all_locations = [
+        {
+            'loc_name': key,
+            'percent': value['in_day'] * 100 / (value['all'] or 1)
+        }
+        for key, value in six.iteritems(best_worst)
+    ]
+
+    all_locations_sorted_by_name = sorted(all_locations, key=lambda x: x['loc_name'])
+    all_locations_sorted_by_percent_and_name = sorted(
+        all_locations_sorted_by_name, key=lambda x: x['percent'], reverse=True)
```

Comment:
I think you can avoid the second sort if you change the key of the first one to `key=lambda x: (x['percent'], x['loc_name'])`. Sorting is relatively expensive (O(n log n)), so probably a good idea.

Reply:
I was hoping to get to this, but it looks unlikely now.